### PR TITLE
[hma][api] Add VPC support to HMA API 

### PR DESCRIPTION
**Pull request by BarrettOlson**
Summary
---------

Builds off of #819 (so only the top three commits should be new with the two commit below that being fixes from 819)

Adds 3 new tfvars:
- `vpc_id` : id of vpc which to restrict the API.
-  `vpc_subnets` : list of private subnets to connect to the `aws_vpc_endpoint` used to access the private API
- `security_groups` : list of security groups to connect to the `aws_vpc_endpoint` used to access the private API

This also adds `examples/vpc_deploy/` to hma to make testing of these changes possible.

Test Plan
---------

Made sure deploy and unit tests still worked without passing in the new arguments. (i.e. does not break existing)
Also ran 30 min soak test that seemed to behave normally. 

Used `examples/vpc_deploy/terraform` + other steps documented in README.md to create a VPC and VPN 
passed output of that to the root terraform and made sure the API was only reachable within the VPN.

### CODE REVIEW

File: hasher-matcher-actioner/terraform/variables.tf [hunk 1]
```diff
@@ -82,6 +82,12 @@ variable "set_sqs_windows_to_min" {
   default     = false
 }
 
+variable "api_in_vpc" {
+  description = "Instead of the usual API establish a private one in a VPC."
```
Comment (schatten) [inline hasher-matcher-actioner/terraform/variables.tf]: usual **public** API... ?

File: hasher-matcher-actioner/terraform/api/main.tf [hunk 2]
```diff
@@ -270,8 +270,10 @@ data "aws_iam_policy_document" "apigateway_assume_role" {
 resource "aws_api_gateway_rest_api" "hma_api_gw" {
   name = "${var.prefix}_hma_api_gw"
   endpoint_configuration {
-    types = ["REGIONAL"]
+    types            = var.vpc_id != "" ? ["PRIVATE"] : ["REGIONAL"]
```
[the remark this answers is not in the document]
Comment (schatten) [reply hasher-matcher-actioner/terraform/api/main.tf:273]: I see you've removed that in the variables. So I take it back. This works too.

[the remark this answers is not in the document]
Comment (BarrettOlson) [reply hasher-matcher-actioner/terraform/api/main.tf:273]: offline sync, having a required bool at the top level is probably best, will add.

**Comment by BarrettOlson** [comment]
Updated package.json. Also @schatten and I noticed the dashboard page was being inconsistent so hiding that page until it can be fixed.

**Pull request merged**